primefaces/src/main/java/org/primefaces/component/selectoneradio/SelectOneRadio.java

```diff
@@ -44,19 +45,7 @@ public class SelectOneRadio extends SelectOneRadioBase {
     public static final String STYLE_CLASS = "ui-selectoneradio ui-widget";
     public static final String NATIVE_STYLE_CLASS = "ui-selectoneradio ui-selectoneradio-native ui-widget";
 
-    private int index = -1;
-
-    public String getRadioButtonId(FacesContext context) {
-        index++;
-
+    public String getRadioButtonId(FacesContext context, int index) {
         return this.getClientId(context) + UINamingContainer.getSeparatorChar(context) + index;
     }
-
-    @Override
-    public Object saveState(FacesContext context) {
```

Comment:
Please explain this
This will likely destroy viewpooling

Reply:
I removed the index, so all that was left was a call to super